bee-ledger/src/workers/snapshot/worker.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::workers::{
+    error::Error,
+    snapshot::{config::SnapshotConfig, error::Error as SnapshotError, import::import_snapshots},
+    storage::{self, StorageBackend},
+};
+
+use bee_message::milestone::MilestoneIndex;
+use bee_runtime::{node::Node, worker::Worker};
+use bee_storage::{access::AsStream, backend::StorageBackend as _, health::StorageHealth};
+use bee_tangle::{solid_entry_point::SolidEntryPoint, MsTangle, TangleWorker};
+
+use async_trait::async_trait;
+
+use chrono::{offset::TimeZone, Utc};
+use futures::stream::StreamExt;
+use log::info;
+
+use std::{any::TypeId, collections::HashMap};
+
+pub struct SnapshotWorker {}
+
+#[async_trait]
+impl<N: Node> Worker<N> for SnapshotWorker
+where
+    N::Backend: StorageBackend,
+{
+    type Config = (u64, SnapshotConfig);
+    type Error = Error;
+
+    fn dependencies() -> &'static [TypeId] {
+        vec![TypeId::of::<TangleWorker>()].leak()
+    }
+
+    async fn start(node: &mut N, config: Self::Config) -> Result<Self, Self::Error> {
+        let (network_id, snapshot_config) = config;
+        let tangle = node.resource::<MsTangle<N::Backend>>();
+        let storage = node.storage();
+
+        match storage::fetch_snapshot_info(&*storage).await? {
+            None => {
```

Comment:
maybe consider using `if let Some(..) = ... { ... } else { ... }` here as it reduces a bit of noise, and puts the happy path (we are happy, right?) first.